protocol/swap/manager.go

```diff
@@ -138,6 +145,19 @@ func (m *manager) GetOngoingSwap(id types.Hash) (*Info, error) {
 	return s, nil
 }
 
+// GetOngoingSwaps returns all ongoing swaps.
+func (m *manager) GetOngoingSwaps() ([]*Info, error) {
+	m.RLock()
+	defer m.RUnlock()
```

Comment:
Is it safe that we return pointers to the `Info` structs both here and in `GetOngoingSwap`? We don't have any synchronization primitives inside the `Info` struct and its status field gets changed over time. Is there any risk of 2 go routines accessing the same `Info` structure?  If one go routine modifies the status and the other is reading the status, the second one may not see the changes in a timely manner. My question is not related to the locking here on the `ongoing` map, but looking at the locks made me think of it.

Reply:
I want to say this is *probably* okay since `GetOngoingSwaps` is only called when the instances start up, and by the RPC server. but I'll change it to be safe